### PR TITLE
frontend Chat component

### PR DESCRIPTION
**Pull request by vaasuu**
Colors might need some changing. 
If you want to test it currently, it's at http://localhost:5173/chat, but it has a hardcoded token and friend ID that you will need to change in ChatDevPage.jsx to match what's in the DB. I'll remove the ChatDevPage stuff in a later PR where I add the chat tab.

The image has everything enabled at once that's why it's showing both event name and friend name on the top of the screen:

localhost_5173_chat (1)

### CODE REVIEW

File: frontend/src/components/chat/Chat.jsx [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+          }}
+        >
+          <input
+            data-testid="message-box"
+            type="text"
+            name="message"
+            disabled={isDisabled}
+            className="chat-container__bottom-area__message-input"
+          />
+          <button
```
Comment (JuhoBjn) [inline frontend/src/components/chat/Chat.jsx:197]: Perhaps we could use the custom button component here to have the styling match the rest of the site. Alternatively, you can copy the styling to this button to make it look the same.

File: frontend/src/components/chat/Chat.css [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  display: flex;
+  flex-direction: row;
+  flex-wrap: nowrap;
+  width: 100%;
+}
+
+.chat-container__bottom-area__message-input {
+  padding: 10px;
+  box-sizing: border-box;
+  width: 100%;
```
Comment (JuhoBjn) [inline frontend/src/components/chat/Chat.css:68]:
```suggestion
  width: 100%;
  background-color: var(--light-green);
  border-style: none;
  font-size: large;
```

The background could be set to light green and set borders to none in order to match the Figma prototype. I think font size should also be set to large.

File: frontend/src/components/chat/Chat.jsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+            e.preventDefault();
+            const message = e.target.message.value;
+            sendMessage(message);
+            e.target.message.value = "";
+          }}
+        >
+          <input
+            data-testid="message-box"
+            type="text"
+            name="message"
```
Comment (JuhoBjn) [inline frontend/src/components/chat/Chat.jsx:193]:
```suggestion
            name="message"
            placeholder="Send a message"
```

A placeholder message could be added here to match the Figma prototype.

File: frontend/src/components/chat/MessageBubble.css [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  flex-direction: column;
+  align-items: flex-end;
+}
+
+.chat-message-bubble__elements--theirs .chat-message-bubble__content {
+  background: var(--gray);
+}
+
+.chat-message-bubble__elements--mine .chat-message-bubble__content {
+  background: var(--blue);
```
Comment (JuhoBjn) [inline frontend/src/components/chat/MessageBubble.css:33]:
```suggestion
  background: var(--light-green);
```

I think we could use light green for the user's own message bubble in order to have the color scheme match the rest of the site.

**Review by JuhoBjn — CHANGES_REQUESTED** [review]
Good work! Functions as intended. I left a couple suggestions regarding the styling of the component in the comments.

**Comment by vaasuu** [comment]
Now looks like this:
chat with improved styles

**Review by JuhoBjn — APPROVED** [review]

**Pull request merged**